### PR TITLE
ENG-1106 (part 2/2): 

### PR DESCRIPTION
**Pull request by daabr**
Continuation of https://github.com/autokitteh/autokitteh/pull/447

All integrations now report connection init successes and failures in a standard way, and the display of this status depends on the init request's origin.

- `/connections/CID/init/vscode` --> VS Code URL
- `/connections/CID/init/web` --> JSON message with "status" (200, 400, 500, etc.) and optional "error" string
- Everything else (CLI, web dashboard) - as usual

The core of this PR is in these files:

- internal/backend/dashboardsvc/connections.go
- internal/backend/oauth/webhook.go
- proto/autokitteh/oauth/v1/svc.proto
- sdk/sdkintegrations/initconn.go

### CODE REVIEW

File: sdk/sdkintegrations/initconn.go [hunk 1]
```diff
@@ -5,37 +5,86 @@ import (
...
+	case "vscode":
+		u := "vscode://autokitteh.autokitteh?cid=%s&status=%d&error=%s"
+		u = fmt.Sprintf(u, c.ConnectionID, status, url.QueryEscape(err))
+		http.Redirect(c.Writer, c.Request, u, http.StatusFound)
+	case "web":
+		u := "/connections/%s/result?status=%d&error=%s"
+		u = fmt.Sprintf(u, c.ConnectionID, status, url.QueryEscape(err))
+		http.Redirect(c.Writer, c.Request, u, http.StatusFound)
+	default: // Local server ("cli", "dash", etc.)
+		path := c.Integration.ConnectionURL().Path + "/error.html"
```
Comment (itayd) [inline sdk/sdkintegrations/initconn.go]: You need to put the right status code here, not just found. If there is an error, the right error code should be returned. 
Also - how do you make Ronen parse the error?

Comment (daabr) [reply sdk/sdkintegrations/initconn.go]: The behavior that Ronen asked for was to always get HTTP 200, and parse the status from a JSON body - but you're right. I confirmed with Ronen that HTTP status + error message in the HTTP response's body is in fact better.

Comment (itayd) [reply sdk/sdkintegrations/initconn.go]: No, we need to return proper status codes. I also told this to Ronen.

File: sdk/sdkintegrations/initconn.go [hunk 2]
```diff
@@ -5,37 +5,84 @@ import (
...
+}
+
+// Abort aborts the connection initialization flow due to
+// a runtime error. It encodes the error status and message,
+// and redirects the user to the last HTTP response, based on its
+// origin (local AK server / local VS Code extension / SaaS web UI).
+func (c ConnectionInit) AbortWithStatus(status int, err string) {
+	switch c.Request.FormValue("origin") {
+	case "vscode":
+		u := "vscode://autokitteh.autokitteh?cid=%s&status=%d&error=%s"
```
Comment (itayd) [inline sdk/sdkintegrations/initconn.go:56]: did you coordinate with ronen re this? not sure though if we shouldn't just error out like before here - does it matter if it's in vscode or not?

File: sdk/sdkintegrations/initconn.go (hunk 1 above)
Comment (daabr) [reply sdk/sdkintegrations/initconn.go]: Yes - already done, see line 59

File: sdk/sdkintegrations/initconn.go (hunk 2 above)
Comment (daabr) [reply sdk/sdkintegrations/initconn.go:56]: Yes, discussed this specific detail this morning

**Review by itayd — APPROVED** [review]

**Pull request merged**